src/ast/irbuilderbpf.h

```diff
@@ -41,7 +41,7 @@ class IRBuilderBPF : public IRBuilder<> {
   AllocaInst *CreateAllocaBPF(int bytes, const std::string &name = "");
   void CreateMemsetBPF(Value *ptr, Value *val, uint32_t size);
   void CreateMemcpyBPF(Value *dst, Value *src, uint32_t size);
-  llvm::Type *GetType(const SizedType &stype);
```

Comment:
Might be good to leave some of the commit message about meaning of `emit_codegen_types` as a comment in here. Not obvious to me from code alone what the param does.

And I'd suggest rewording:

> we want to use i64 for pointers and byte arrays for structs as that's how we actually store them,

to something about how it's what libbpf expects. That's the real reason, right?

Reply:
> Might be good to leave some of the commit message about meaning of `emit_codegen_types` as a comment in here. Not obvious to me from code alone what the param does.

Good idea, will do.


> And I'd suggest rewording:
> 
> 
> 
> > we want to use i64 for pointers and byte arrays for structs as that's how we actually store them,
> 
> 
> 
> to something about how it's what libbpf expects. That's the real reason, right?
> 

I don't think so. IIUC, we store pointers as ints and structs as byte arrays b/c it's easier to store them in maps that way (maybe BPF maps didn't even allow storing raw pointers in maps back in the time when bpftrace was created).

OTOH, libbpf requires kfunc declarations to have types matching those in the kernel which is why I had to introduce the `emit_codegen_types` param. I'll try to reword it in the comment and commit message to make it clearer.

Reply:
Updated both the commit message and the relevant function comments.